### PR TITLE
CO-1187: Latest hive doesn't honor syncset-pause annotation

### PR DESCRIPTION
**Pull request by abutcher**
/assign @staebler 
/cc @dgoodwin

### CODE REVIEW

File: pkg/controller/clustersync/clustersync_controller.go [hunk 1]
```diff
@@ -282,6 +282,11 @@ func (r *ReconcileClusterSync) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	// Is syncing paused?
```
Comment (2uasimojo) [inline pkg/controller/clustersync/clustersync_controller.go:285]: Should we consider putting this [after creating the initial ClusterSync and ClusterSyncLease](https://github.com/openshift/hive/pull/1147/files#diff-d64ff608a67db50bd6a15a35465607c1R341)?

File: pkg/controller/clustersync/clustersync_controller_test.go [hunk 2]
```diff
@@ -278,6 +278,10 @@ func TestReconcileClusterSync_NoWorkToDo(t *testing.T) {
 				}),
 			),
 		},
+		{
+			name: "syncset pause",
+			cd:   cdBuilder(scheme).GenericOptions(testgeneric.WithAnnotation(constants.SyncsetPauseAnnotation, "true")).Build(),
+		},
```
Comment (2uasimojo) [inline pkg/controller/clustersync/clustersync_controller_test.go:284]: Confirmed this test fails when the fix is reverted ✓

**Review by 2uasimojo — COMMENTED** [review]
Fix looks good. One question/suggestion inline. I shall avoid official lgtm tag in any case, because that seems to merge things _sometimes_.

File: pkg/controller/clustersync/clustersync_controller.go (hunk 1 above)
Comment (dgoodwin) [reply pkg/controller/clustersync/clustersync_controller.go:285]: This looks consistent with what would happen today if the cluster was marked unreachable. I can't think of a reason we'd need a clustersync created for a new cluster that was immediately paused.

**Comment by 2uasimojo** [comment]
Is there a way to stop this burning CI resources until the issue is resolved?

**Comment by staebler** [comment]
> Is there a way to stop this burning CI resources until the issue is resolved?

/hold

**Comment by staebler** [comment]
/hold cancel

Verify tests have been fixed.

**Pull request merged**